### DISCUSSION TITLE
How to authenticate Octokit via the GitHub CLI?

### DISCUSSION THREAD

**Discussion by electrovir** [octokit/octokit.js#2614] [Q&A]
I'm trying to use Octokit for a TypeScript script that I want to run locally on my own computer (only).

I already have the GitHub CLI installed (`gh`) and authenticated. How do I utilize that already-authenticated service to authenticate an Octokit class instance?

I could just execute raw shell commands from my script (like `gh pr list`) but then I get none of the Octokit types or convenient API.

**Comment by pauldraper** [comment]
```sh
gh auth token
```
